compiler/rustc_smir/src/rustc_internal/mod.rs

```diff
@@ -20,7 +21,7 @@ impl<'tcx> Index<stable_mir::DefId> for Tables<'tcx> {
 
     #[inline(always)]
     fn index(&self, index: stable_mir::DefId) -> &Self::Output {
-        &self.def_ids[index.0]
+        &self.def_ids.get_index(index.0).unwrap().0
```

Comment:
would be nice if we debug-assert that the value is the same index that we started with.

```suggestion
        let (k, v) = self.def_ids.get_index(index.0).unwrap();
        assert_eq!(*v, index);
        k
```

same for all of these

Reply:
I think you meant comparing key and index here, but yeah I agree with you it's a nice safety feature.

```
assert_eq!(*k, index);
```

Reply:
ahh sorry, what you wrote is correct I misread my own code 🤦